File: src/com/company/XFileReader.java
package com.company;

import org.apache.poi.xssf.usermodel.XSSFWorkbook;

import java.io.File;
import java.io.FileInputStream;

/**
 * Performs operations with excel files.
 */
public class XFileReader {
    /**
     * Loads data from a given excel file
     *
     * @param filePath a path to the file to read from
     * @return
     */
    public XSSFWorkbook loadFromFile(final String filePath) {
        XSSFWorkbook workbook = null;
        try {
            File f = new File(filePath);
            FileInputStream file = new FileInputStream(f);
            workbook = new XSSFWorkbook(file);
            file.close();
        } catch (Exception e) {
            System.out.println("Error " + e.getMessage() + " when processing file " + filePath);
            System.out.println("Try to open the file by Excel and save it with extension .xlsx");
        }
        if (workbook != null) {
            System.out.println("Loaded " + workbook.getSheetAt(0).getPhysicalNumberOfRows() + " rows from file " + filePath);
        }
        return workbook;
    }


}

File: src/com/company/XFileReader.kt
package com.company

import org.apache.poi.xssf.usermodel.XSSFWorkbook

import java.io.File
import java.io.FileInputStream

/**
 * Performs operations with excel files.
 */
class XFileReader {
    /**
     * Read an Excel file and return it as Workbook instance.

     * @param filePath a path to the file to read from
     * *
     * @return a XSSWorkbook instance
     */
    fun loadFromFile(filePath: String): XSSFWorkbook {
        val f = File(filePath)
        val file = FileInputStream(f)
        val workbook = XSSFWorkbook(file)
        println("Loaded " + workbook.getSheetAt(0).physicalNumberOfRows + " rows from file " + filePath)
        file.close()
        return workbook
    }


}
